ops/ruby/devops-toolkit/tasks/monitoring_assets.rb

```diff
@@ -1,65 +1,227 @@
+      logger.info("found  #{custom_metrics.count} custom metrics")
       counts['custom_metrics'] = custom_metrics.count
       custom_metrics.each do |metric|
-        @logger.info("\t#{metric.name}: #{metric.description}")
+        logger.info("\t#{metric.name}: #{metric.description}")
       end
 
-      user_logs_based_metrics = metric_client.list_metric_descriptors(metric_project_path, {filter: USER_LOGS_BASED_METRIC_FILTER})
+      user_logs_based_metrics = @metric_client.list_metric_descriptors(@project_path, {filter: USER_LOGS_BASED_METRIC_FILTER})
       counts['user_logs_based_metrics'] = user_logs_based_metrics.count
-      @logger.info("found  #{user_logs_based_metrics.count} user-defined  logs-based metrics")
+      logger.info("found  #{user_logs_based_metrics.count} user-defined  logs-based metrics")
 
       user_logs_based_metrics.each do |metric|
-        @logger.info("\t#{metric.name}: #{metric.description}")
+        logger.info("\t#{metric.name}: #{metric.description}")
       end
 
-      total_logs_based_metrics = metric_client.list_metric_descriptors(metric_project_path, {filter: LOGS_BASED_METRIC_FILTER})
-      @logger.info("found  #{total_logs_based_metrics.count} total logs-based metrics")
+      total_logs_based_metrics = metric_client.list_metric_descriptors(@project_path, {filter: LOGS_BASED_METRIC_FILTER})
+      logger.info("found  #{total_logs_based_metrics.count} total logs-based metrics")
 
-      alerts_client = Google::Cloud::Monitoring::V3::AlertPolicyServiceClient.new
-      policies = alerts_client.list_alert_policies(env.formatted_project_name)
-      @logger.info("found #{policies.count} alerting policies")
+      policies = @alert_client.list_alert_policies(@project_path)
+      logger.info("found #{policies.count} alerting policies")
       counts['policies'] = policies.count
 
       policies.each do |policy|
-        @logger.info("\t#{policy.display_name}, #{policy.name}, #{policy.conditions.count} conditions")
+        logger.info("\t#{policy.display_name}, #{policy.name}, #{policy.conditions.count} conditions")
       end
 
-      @logger.info("Total counts for #{env.short_name}: #{counts}")
+      logger.info("Total counts for #{@current_env.short_name}: #{counts}")
+    end
+  end
+
+  DO_BACKUP_ALL = false # TODO(jaycarlton): make this an option
+
+  # For every environment in the envs_file, fetch all items of interest from the appropriate
+  # Stackdriver APIs. (Note that in future we may wish to broaden this to other )
+  def backup_config
+    visit_envs do |env_bundle|
+      backup_alert_policies
+      if DO_BACKUP_ALL
+        backup_all_metrics
+      else
+        backup_custom_metrics
+        backup_user_logs_based_metrics
+      end
+      backup_dashboards
+      backup_group_members
+      backup_monitored_resources
+      backup_notification_channels
+      backup_notification_channels
```

Comment:
I assume adding this twice was not intentional

Reply:
good catch